compiler/daml-lf-reader/src/DA/Daml/LF/Reader.hs

```diff
@@ -111,13 +121,27 @@ readDalfManifest dar = do
         lookup attrName attrs
     missingAttr attrName = Left $ "No " <> BSUTF8.toString attrName <> " attribute in manifest."
 
-readDalfs :: Archive -> Either String Dalfs
+readDalfs :: Archive -> Either String (Dalfs BSL.ByteString)
 readDalfs dar = do
     DalfManifest{..} <- readDalfManifest dar
     mainDalf <- getEntry dar mainDalfPath
     dalfs <- mapM (getEntry dar) (delete mainDalfPath dalfPaths)
     pure $ Dalfs mainDalf dalfs
 
+extractNameAndPackageIdFromPath :: FilePath -> (T.Text, LF.PackageId)
+extractNameAndPackageIdFromPath = bimap (T.pack . intercalate "-") (LF.PackageId . T.pack) . fromJust . unsnoc . wordsBy (=='-') . takeBaseName
```

Comment:
unrelated to the other thread, instead of splitting and then rejoining by hyphens, you could `T.pack` the basename and then `T.breakOnEnd` to split the package name and package id